### PR TITLE
fixes daily-build regressions related to npm scripts migration

### PR DESCRIPTION
**Pull request by mshima**
Fixes daily-build regressions related to npm scripts migration.

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] The JDL part is updated if necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/index.js [hunk 1]
```diff
@@ -459,6 +459,11 @@ module.exports = class JHipsterServerGenerator extends BaseBlueprintGenerator {
               'docker:db:await': `echo "Waiting for Couchbase to start" && wait-on -t ${WAIT_TIMEOUT} http-get://localhost:8091/ui/index.html && sleep 30 && echo "Couchbase started"`,
             });
           }
+          if (prodDatabaseType === MYSQL) {
```
Comment (Tcharl) [inline generators/server/index.js:462]: I guess that now we should also add this line if devDatabaseType is Mysql too.
Maybe a liveliness probe in the docker-compose file and in the TestContainer implementation would be more efficient than an additional command

Comment (mshima) [reply generators/server/index.js]: Testcontainer doesn’t have this problem.
Docker compose uses sleep at microservices.

A liveliness in entrypoint is nice but can be trick.

Comment (Tcharl) [reply generators/server/index.js]: I remember that I made some liveliness probes for some jh services in one of my previous project, let me check

File: generators/server/index.js [hunk 2]
```diff
@@ -490,13 +495,20 @@ module.exports = class JHipsterServerGenerator extends BaseBlueprintGenerator {
               scriptsStorage.set(`docker:${dockerConfig}:build`, `docker-compose -f ${dockerFile} build`);
               dockerBuild.push(`npm run docker:${dockerConfig}:build`);
             } else if (dockerConfig === 'jhipster-registry') {
-              dockerAwaitScripts.push(
+              if (authenticationTypeOauth2) {
+                dockerOthersUp.push('npm run docker:keycloak:await');
+              }
+              scriptsStorage.set(
+                `docker:jhipster-registry:await`,
```
Comment (DanielFran) [inline generators/server/index.js:502]:
```suggestion
                'docker:jhipster-registry:await',
```

**Review by DanielFran — CHANGES_REQUESTED** [review]

**Pull request merged**